### PR TITLE
Cap fire age to prevent android overflow

### LINKED ISSUE #33130: Integer Overflow on fire's duration
Integer Overflow on fire's duration

# Steps To Reproduce

1. Put things in brazier
2. Light fire
3. Examine fire after a while

# Expected behavior

On Android experimental builds fires don't behave properly. They go out with material unburned. Adding fuel to a fire does nothing. Firewood source does not prevent fire from going out during crafting. 

# Screenshots

https://i.redd.it/i7y5jqq0jub31.jpg

# Versions and configuration

 - - OS: Android
    - OS Version: Manufacturer: Q-Innovations; Model: QS5509A; Release: 8.1.0; Incremental: v1A71-0; 
- Game Version: 0.D-6156-gcb9471d [32-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    C.R.I.T Expansion Mod [crt_expansion],
    Bens GF recipes [Tolerate_This],
    Craftable Gun Pack [craftgp],
    DeadLeaves' Fictional Guns [FIC_Weapons],
    Garden Pots [growable-pots],
    Icecoon's Arsenal [ew_pack],
    Makeshift Items Mod [makeshift],
    Medieval and Historic Classes and Shields [Medieval_Stuff],
    More Survival Tools [More_Survival_Tools],
    DinoMod [DinoMod],
    Modular Turrets [modular_turrets],
    Salvaged Robots [Salvaged_Robots],
    Crazy Cataclysm [crazy_cataclysm],
    Hydroponics [hydroponics],
    Mutant NPCs [mutant_npcs],
    Beta National Guard Camp [national_guard_camp],
    More Locations [more_locations],
    Fuji's More Buildings [FujiStruct],
    Mapgen Demo [mapgen_demo],
    No Rail Stations [No_Rail_Stations],
    Urban Development [Urban_Development],
    Folding Parts pack [deoxymod],
    Vehicle Additions Pack [blazemod],
    Tanks and Other Vehicles [Tanks],
    Bionic Slots [cbm_slots],
    Classes and Scenarios Mod [more_classes_scenarios],
    Sleep Deprivation [sleepdeprivation],
    SpeedyDex [speedydex],
    StatsThroughSkills [StatsThroughSkills]
]

# Additional context

Issue also reported in Reddit thread which screenshot is taken from
https://www.reddit.com/r/cataclysmdda/comments/cgceex/uh_how_long_exactly/

Also 

https://www.reddit.com/r/cataclysmdda/comments/cl609h/i_uhhhhh_guess_thats_good/

Seems related

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Bugfixes "Cap fire age to prevent android overflow"```

#### Purpose of change
Fixes #33130 

#### Describe the solution
cap fire age as per Kevins comment in linked issue.

#### Describe alternatives you've considered
N/A

#### Testing
I have no android device to test this on, so leaving this here as a draft proposal to see if others can test it and it fixes the issue.
#### Additional context
I dont know if there are other places in the code that add to field age, so that may need to be addressed too.

### CODE REVIEW

**Comment by claudiorodriguez** [comment]
This was also happening on OSX, at least on 10.14. Not very familiar with the build process to test the fix though

**Comment by anothersimulacrum** [comment]
If you're interested in trying, there's [instructions for OSX](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/COMPILING/COMPILING.md#advanced-info-for-developers).

File: src/map_field.cpp [hunk 1]
```diff
@@ -476,6 +476,7 @@ bool map::process_fields_in_submap( submap *const current_submap,
                     sblk.apply_slime( p, cur.get_field_intensity() * curtype.obj().apply_slime_factor );
                 }
                 if( curtype == fd_fire ) {
+                    cur.set_field_age( std::min( 24_hours, cur.get_field_age() ) );
```
Comment (kevingranade) [inline src/map_field.cpp]: Kinda dumb thing about fire "age", it's negative.
If a fire is "10 minutes old" it's at great risk of going out, if it has a negative age it's in no risk of going out.
So this needs to cap at -24_hours

Comment (davidpwbrown) [reply src/map_field.cpp]: hmm, til.

**Comment by claudiorodriguez** [comment]
I was able to get an OSX build going for this branch, and the issue seems fixed as far as I can tell, this makes the game actually playable again. Thanks!

**Pull request merged**